tests/lava/proc/monitor/test_monitors.py

```diff
@@ -0,0 +1,288 @@
+import numpy as np
+import unittest
+
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+from lava.magma.core.model.py.ports import PyRefPort, PyVarPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.process.ports.ports import RefPort, VarPort
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.variable import Var
+from lava.magma.core.resources import CPU
+from lava.magma.core.sync.domain import SyncDomain
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.run_configs import RunConfig
+from lava.magma.core.run_conditions import RunSteps
+from lava.magma.core.decorator import implements, requires
+from lava.proc.monitor.process import Monitor
+from lava.proc.monitor.models import PyMonitorModel
+from lava.proc.lif.process import LIF
+from lava.magma.compiler.compiler import Compiler
+from lava.magma.core.run_configs import Loihi1SimCfg
+
+
+class MonitorRunConfig(RunConfig):
+    """
+    The RunConfic class for the unittests
+    """
+    def __init__(self, **kwargs):
+        sync_domains = kwargs.pop("sync_domains")
+        super().__init__(custom_sync_domains=sync_domains)
+        self.model = None
+        if "model" in kwargs:
+            self.model = kwargs.pop("model")
+
+    def select(self, process, proc_models):
+        if self.model is not None:
+            if self.model == "sub" and isinstance(process, AbstractProcess):
+                return proc_models[1]
+        return proc_models[0]
+
+
+class LifRunConfig(RunConfig):
+    """Run configuration selects appropriate LIF ProcessModel based on tag:
+    floating point precision or Loihi bit-accurate fixed point precision"""
+    def __init__(self, custom_sync_domains=None, select_tag='fixed_pt'):
+        super().__init__(custom_sync_domains=custom_sync_domains)
+        self.select_tag = select_tag
+
+    def select(self, proc, proc_models):
+        for pm in proc_models:
+            if self.select_tag in pm.tags:
+                return pm
+        raise AssertionError("No legal ProcessModel found.")
+
+
+class P1(AbstractProcess):
+    """a dummy proc with two Vars"""
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.s = Var(shape=(1,), init=1)
+        self.u = Var(shape=(1,), init=0)
+
+
+# A minimal PyProcModel implementing P1
+@implements(proc=P1, protocol=LoihiProtocol)
+@requires(CPU)
+class PyProcModel1(PyLoihiProcessModel):
+    s: np.ndarray = LavaPyType(np.ndarray, np.int32)
+    u: np.ndarray = LavaPyType(np.ndarray, np.int32)
+
+    def pre_guard(self):
+        return True
+
+    def run_pre_mgmt(self):
+        if self.current_ts > 1:
+            self.s = np.array([self.current_ts])
+            self.u = 2*np.array([self.current_ts])
```

Comment:
I think this is missing white-space around the `*` operator -> `2 * np.array` to avoid linter error